File: FindLongestString.py
# Find Longest String in List
# - Given a list of strings strs, return the longest string.
# - Time: O(n), Space: O(1)

class Solution:
    def findLongestString(self, strs: list[str]) -> str:
        if not strs:
            return ""
        longest = strs[0]
        for s in strs[1:]:
            if len(s) > len(longest):
                longest = s
        return longest

File: FindLongestString.cs
// Find Longest String in List
// - Given a list of strings strs, return the longest string.
// - Time: O(n), Space: O(1)

using System;
using System.Collections.Generic;

class Solution
{
    public string FindLongestString(List<string> strs)
    {
        if (strs == null || strs.Count == 0)
            return "";

        string longest = strs[0];

        foreach (var s in strs)
        {
            if (s.Length > longest.Length)
            {
                longest = s;
            }
        }

        return longest;
    }
}
